Objective-C/CBLReplicator.mm

```diff
@@ -565,9 +565,12 @@ - (void) onDocsEnded: (NSArray<CBLReplicatedDocument*>*)docs pushing: (BOOL)push
                                                           error: &error]) {
                     [doc resetError]; // Reset the error as successfully resolving the conflicts
                 } else {
-                    // TODO: Handle the errors when the conflict happens, with the pull
                     CBLWarn(Sync, @"%@: Conflict resolution of '%@' failed: %@",
                             self, doc.id, error);
+                    id replication = [[CBLDocumentReplication alloc] initWithReplicator: self
+                                                                                 isPush: pushing
+                                                                              documents: @[doc]];
+                    [_docReplicationNotifier postChange: replication];
```

Comment:
I'm not quite sure about the logic here as the ReplicatedDocument has already been notified with the conflict error. Now it will get notified again with another error.

I think we need to revise the whole logic around here. It could be that we will not notify ReplicatedDocument until the conflict gets resolved or has an error. Can you implement this logic and create a new issue to review the logic?

Reply:
I can add a separate PR for whole error handling. keeping this same logic with a TODO comment saying to revert this.

Reply:
https://github.com/couchbase/couchbase-lite-ios/issues/2444